finals-2024-01-16/code/5140-simona-blazevska/src/app/band-detail/band-detail.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Band } from '../models/bands';
+import { HttpClient } from '@angular/common/http';
+
+@Component({
+  selector: 'app-band-detail',
+  templateUrl: './band-detail.component.html',
+  styleUrls: ['./band-detail.component.css'],
+})
+export class BandDetailComponent implements OnInit {
+  band: Band | undefined;
+  yearsActive: string | undefined;
+  similarBands: Band[] = [];
+
+  constructor(private route: ActivatedRoute, private http: HttpClient) {}
+
+  ngOnInit(): void {
+    this.route.params.subscribe((params) => {
+      const bandId = +params['id']; // Convert the ID to a number
+      this.loadBandDetails(bandId);
+    });
+  }
+
+  loadBandDetails(bandId: number): void {
+    // Fetch band details based on the ID
+    const url = `http://localhost:3000/bands/${bandId}`;
+
+    this.http.get<Band>(url).subscribe((data) => {
```

Comment:
The point of the services approach is to decouple the facts of where the data comes from, and how is it used. That is missing here.